src/main/java/gregtech/api/recipes/machines/RecipeMapAssemblyLine.java

```diff
@@ -5,13 +5,23 @@
 import gregtech.api.gui.ModularUI;
 import gregtech.api.gui.widgets.ProgressWidget;
 import gregtech.api.gui.widgets.SlotWidget;
+import gregtech.api.recipes.Recipe;
 import gregtech.api.recipes.RecipeBuilder;
 import gregtech.api.recipes.RecipeMap;
+import gregtech.api.recipes.recipeproperties.ResearchProperty;
+import it.unimi.dsi.fastutil.objects.Object2ObjectOpenHashMap;
+import it.unimi.dsi.fastutil.objects.ObjectOpenHashSet;
 import net.minecraftforge.items.IItemHandlerModifiable;
 
 import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
 
-public class RecipeMapAssemblyLine<R extends RecipeBuilder<R>> extends RecipeMap<R> {
+public class RecipeMapAssemblyLine<R extends RecipeBuilder<R>> extends RecipeMap<R> implements IResearchRecipeMap {
+
+    /** Contains the recipes for each research key */
+    private final Map<String, Collection<Recipe>> researchEntries = new Object2ObjectOpenHashMap<>();
```

Comment:
Yes. The idea with this system is that you could make one scan produce any amount of recipes. This would allow better customization on research costs and the process overall. Often players find it monotonous to scan every component just to get the next tier's recipe over and over, for example.